web/modules/custom/mymodule/src/Plugin/Block/Stockblock.php

```diff
@@ -0,0 +1,139 @@
+<?php
+
+namespace Drupal\mymodule\Plugin\Block;
+
+use Drupal\Core\Block\BlockBase;
+use Drupal\Core\Block\BlockPluginInterface;
+use Drupal\Core\Form\FormStateInterface;
+use GuzzleHttp\Client;
+
+/**
+ * Provides a 'Stock' Block.
+ *
+ * @Block(
+ *   id = "stock_block",
+ *   admin_label = @Translation("Stock block"),
+ * )
+ */
+class Stockblock extends BlockBase implements BlockPluginInterface {
+
+  /**
+   * Method to return the values to block.
+   *
+   * @return array
+   *   It will return the array to render elements.
+   */
+  public function build() {
+    $config = $this->getConfiguration();
+
+    $company_symbol = $config['company_symbol'];
+    $start_date = empty($config['start_date']) ? date('Y-m-d') : $config['start_date'];
+    $end_date = empty($config['end_date']) ? date('Y-m-d') : $config['end_date'];
+    // $start_date ='2022-2-15';
+    // $end_date = '2022-2-15';
+    $service_url = "https://api.tiingo.com/tiingo/daily/" . $company_symbol . "?token=34f412d51db4046a81f4180aad2233c41df5d3b1";
```

Comment:
@anitha-innoraft - Let's create a constant for `"https://api.tiingo.com/tiingo/daily/"`

Reply:
Created the constant.